autoscoper/src/ui/AutoscoperMainWindow.cpp

```diff
@@ -2101,38 +2101,65 @@ Autoscoper 1 was developed by Andy Loomis(original CUDA version) and Mark Howiso
   about_autoscoper->show();
 }
 
+bool AutoscoperMainWindow::isInstalled() const {
+    // check if autoscoper was downloaded from the slicer extension manager or if it was built from source
+    // If it is installed the executable should be in the SlicerAutoscoperM/bin folder
+
+    QString root_path = QCoreApplication::applicationDirPath();
+    QStringList path_list = root_path.split("/");
+    std::reverse(path_list.begin(), path_list.end());
+    return path_list.at(1) == "SlicerAutoscoperM";
+}
+
+QString AutoscoperMainWindow::get_sample_data_path(const QString& default_config_path, const QString& sample_type) {
+    if (!isInstalled()) // If the application is not installed (ie. it's built), use the root path (default value for default_config_path)
+        return default_config_path;
+    
+    // If the application is installed, use the Slicer cache folder
+#ifdef WIN32
+    QString slicer_cache_dir = QDir::homePath() + "/AppData/Local/slicer.org/Slicer/cache/SlicerIO/" + sample_type;
+#else
+    QString slicer_cache_dir = QDir::homePath() + "/.config/slicer.org/Slicer/cache/SlicerIO/" + sample_type;
+#endif
```

Comment:
```suggestion
    QString app_data_dir = QStandardPaths::writableLocation(QStandardPaths::AppLocalDataLocation);
    QString slicer_cache_dir =  "%1/cache/SlicerIO/%2").arg(app_data_dir, sample_type);
```

Reply:
I think `CacheLocation` may be more appropriate.

In that case, the following should be done:

```
    QString cache_location = QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
    QString cache_path = QFileInfo(cache_location, this->applicationName() + "IO").absoluteFilePath();
```

I think any attempt to guess the `Slicer` cache path would be impossible as the user can change it in the settings.


See https://github.com/Slicer/Slicer/blob/8d9e99fbc7e41c712b4d956db7b804b580c825be/Base/QTCore/qSlicerCoreApplication.cxx#L1411-L1439